### PR TITLE
Fix long line matches

### PR DESCRIPTION
**Pull request by Numkil**
This pull requests resolve problem #95.

I ask ag to only display the match and nothing else of the string. Then I construct a regex allowing up to 50 characters before and 50 characters behind to be included in the matching result.

Let me know what you think. :)

### CODE REVIEW

**Comment by Numkil** [comment]
I just noticed that -o and --vimgrep do not work together, the -o switch is ignored. I will contact the_silver_searcher about this.

**Comment by losingkeys** [comment]
Great idea! Unfortunately I'm having some issues with the branch. When I search [my dotfiles](https://github.com/losingkeys/dotfiles) for `set`, it says "no matches", but when I switch back to the master branch, it shows matches (some with annoyingly long lines).

Another thought: will this work for all window sizes? I think it'll still wrap for some windows, which might be ok (because it probably won't wrap that many times), but it'd be nice to get it exact too.

50e30ae is a duplicate from #100 (merged). I can just cherry-pick the other commit when it's ready, or you can rebase the pull, whichever you prefer.

This is another pull that would benefit from tests, gotta be careful when adding regexes to others' search strings :smile:.

**Comment by Iazel** [comment]
You should notice that `grepargs` isn't always the regexp, but could include other options, e.g. I usually include a filetypes restriction, so the result would be something like `.\{0,50}--ft myregexp.\{0,50}`. This also happen if you want to search in a subdir only. I bet @losingkeys has done at least one of these, and that's cause the no-match.

I think the `-o` flag is enough and to not include it by default, let the user decide when he need it or not (like every other options), plus it's not widely supported (same situation of `--vimgrep`).

**Comment by losingkeys** [comment]
Ohh, I didn't think of that. Good catch @lazel. My current `g:ag_prg` is
`ag --vimgrep --smartcase`. It should probably handle other ag options too,
like `--literal`.

**Comment by Numkil** [comment]
Yeah those are some good points, which I didn't think about. And you are right that's I'm sure why it doesn't work for @losingkeys. I had just made this quickly as a proof that it might work, but apparently it doesn't even because -o and --vimgrep don't work together, which is a limitation in ag itself. Maybe just setting nowrap as @Iazel suggested is the only way to go even if it's not perfect.

File: autoload/ag.vim [hunk 1]
```diff
@@ -94,7 +94,7 @@ function! ag#Ag(cmd, args)
     let &grepformat=g:ag_format
     set t_ti=
     set t_te=
-    silent! execute a:cmd . " " . escape(l:grepargs, '|')
+    silent! execute a:cmd . " .\\{0,50}" . escape(l:grepargs, '|').".\\{0,50}"
```
Comment (losingkeys) [inline autoload/ag.vim]: This breaks some searches. For instance, if I run `:Ag set` in my dotfiles, for some reason it doesn't return any matches. However if I run `:Ag -u set` in my dotfiles, I got lots of matches (from ignored files). I don't see any matches with `:Ag -u set` when using this branch.

**Pull request closed without merging**